plugins/module_utils/pulp_helper.py

```diff
@@ -225,7 +231,8 @@ def chunked_upload(self, path, sha256, size):
                         end=offset + actual_chunk_size - 1,
                         size=size,
                     )
-                    with TemporaryDirectory() as temp_dir:
+                    try:
+                        temp_dir = mkdtemp(dir="/tmp")
```

Comment:
I think you need to switch those two lines again.
There is no point in `rmtree(temp_dir)` if `mkdtemp` threw the exception.

Reply:
Good call.